bemb/model/bemb.py

```diff
@@ -1313,18 +1334,21 @@ def elbo(self, batch: ChoiceDataset, num_seeds: int = 1) -> torch.Tensor:
         # ==============================================================================================================
         if self.pred_item:
             # the prediction target is item_index.
-            elbo += self.forward(batch,
+            elbo_expanded = self.forward(batch,
                                  return_type='log_prob',
                                  return_scope='item_index',
                                  deterministic=False,
-                                 sample_dict=sample_dict).sum(dim=1).mean(dim=0)  # (num_seeds, len(batch)) --> scalar.
+                                 sample_dict=sample_dict)
+            if self.deterministic_variational:
+                elbo_expanded = elbo_expanded.unsqueeze(dim=0)
+            elbo += elbo_expanded.sum(dim=1).mean(dim=0)  # (num_seeds, len(batch)) --> scalar.
         else:
             # the prediction target is binary.
             # TODO: update the prediction function.
             utility = self.forward(batch,
                                    return_type='utility',
                                    return_scope='item_index',
-                                   deterministic=False,
+                                   deterministic=self.deterministic_variational,
```

Comment:
Do we need to expand to create the seed dimension for the return utility as well if it is deterministic?

Reply:
yes; we should. we are doing that in the sample_coef_dict; this keeps things consistent. I'm submitting a small fix to code to make sure it works right.